docs/api/navigators/StackNavigator.md

```diff
@@ -118,7 +118,7 @@ All `navigationOptions` for the `StackNavigator`:
 
 - `title` - a title (string) of the scene
 - `header` - a config object for the header bar:
-  - `visible` - Boolean toggle of header visibility. Only works when `headerMode` is `screen`.
+  - `bar` - React Element to display as a header or `null` to hide it. Only works when `headerMode` is `screen`
```

Comment:
> One should be able to override header for a single scene without implementing entire header once again

You're overriding entire header in either case. So it's not much different. It's just that you don't need to deal with animations when `mode` is float, so it's simpler.

> However, if we let people go with header.bar, there's no need to keep headerComponent I believe, isn't it?
     
Yes, that's what I'm saying